crates/uv-fs/src/lib.rs

```diff
@@ -210,23 +213,21 @@ pub async fn rename_with_retry(
         let from = from.as_ref();
         let to = to.as_ref();
 
-        let backoff = backoff_file_move();
-        backoff::future::retry(backoff, || async move {
-            match fs_err::rename(from, to) {
-                Ok(()) => Ok(()),
-                Err(err) if err.kind() == std::io::ErrorKind::PermissionDenied => {
-                    warn!(
-                        "Retrying rename from {} to {} due to transient error: {}",
-                        from.display(),
-                        to.display(),
-                        err
-                    );
-                    Err(backoff::Error::transient(err))
-                }
-                Err(err) => Err(backoff::Error::permanent(err)),
-            }
-        })
-        .await
+        let rename = || async { fs_err::rename(from, to) };
+
+        rename
+            .retry(backoff_file_move())
+            .sleep(tokio::time::sleep)
```

Comment:
I think these sleep calls are technically defaulted but their examples use them and once they were there I felt they improved readability.